tests/cupy_tests/functional_tests/test_piecewise.py

```diff
@@ -0,0 +1,73 @@
+import unittest
+
+import pytest
+
+import cupy
+from cupy import testing
+
+
+class TestPiecewise(unittest.TestCase):
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
```

Comment:
Please test also in bool and complex inputs.
```suggestion
    @testing.for_all_dtypes()
```